ce/insights.ts

```diff
@@ -23,18 +25,47 @@ export const insights = setup('b4e88960-4393-4dd9-ab8e-97e8fe6d7603').
   setSendLiveMetrics(false).
   setUseDiskRetryCaching(false).
   start();
-export function trackEvent(name: string, properties?: { [key: string]: string }) {
+
+defaultClient.context.keys.applicationVersion = Version;
+
+
+// todo: This will be refactored to allow appInsights to be called out-of-proc from the main process.
+//       in order to not potentially slow down or block on activation/etc.
+
+export function flushTelemetry() {
+  session.channels.debug('Ensuring Telemetry data is finished sending.');
+  defaultClient.flush({});
+}
+
+defaultClient.addTelemetryProcessor((envelope, contextObjects) => {
+  if (session.context['printmetrics']) {
+    session.channels.message(`Telemetry Event: \n${JSON.stringify(envelope.data, null, 2)}`);
+  }
+
+  // only actually send telemetry if it's enabled.
+  return session.telemetryEnabled;
+});
+
+export function trackEvent(name: string, properties: { [key: string]: string } = {}) {
+  session.channels.debug(`Triggering Telemetry Event ce.${name}`);
+
   defaultClient.trackEvent({
-    name: 'Microsoft.ApplicationInsights.Event',
-    tagOverrides: {
-      'ai.device.os': process.platform,
-      'ai.device.osVersion': version(),
-    },
+    name: `ce/${name}`,
+    time: new Date(),
+
     properties: {
       ...properties,
```

Comment:
cloning it to ensure that the properties object isn't shared between any event calls (which could happen in the future)